src/Refinement/RefinedDiscreteModels.jl

```diff
@@ -0,0 +1,110 @@
+
+
+"""
+Refinement glue between two nested triangulations
+- `f2c_faces_map`     : Given a fine face gid, returns the gid of the coarse face containing it. 
+- `fcell_to_child_id` : Given a fine cell gid, returns the local child id within the coarse cell containing it.
+- `f2c_ref_cell_map`  : Ref coordinate map between the fcells and their parent ccell, i.e 
+                          for each fine cell defines Φ st. x_c = Φ(x_f)
+"""
+struct RefinementGlue{A,B,C} <: GridapType
+  f2c_faces_map::A
+  fcell_to_child_id::B
+  f2c_ref_cell_map::C
+
+  function RefinementGlue(f2c_faces_map,
+                          fcell_to_child_id,
+                          f2c_ref_cell_map)
+    A = typeof(f2c_faces_map)
+    B = typeof(fcell_to_child_id)
+    C = typeof(f2c_ref_cell_map)
+    new{A,B,C}(f2c_faces_map,fcell_to_child_id,f2c_ref_cell_map)
+  end
+end
+
+"""
+Ref coordinate map from fcell ref coords to ccell ref coords.
+Size -> Number of children per coarse cell.
+"""
+function get_f2c_ref_cell_map(reffe,ref)
+  ref_grid = Geometry.UnstructuredGrid(Visualization.compute_reference_grid(reffe,ref))
+  return get_cell_map(ref_grid)
+end
+
+"""
+Ref coordinate map between fine and coarse cells. 
+Size -> Number of fine cells. 
+"""
+function get_f2c_ref_coordinate_map(g::RefinementGlue)
+  m = Reindex(g.f2c_ref_cell_map)
+  return lazy_map(m,g.fcell_to_child_id)
+end
+
+#################################################################
+
+"""
+  RefinedDiscreteModel
+
+  `DiscreteModel` created by refining another `DiscreteModel`. 
+  The hierarchy is stored, allowing for the transfer of dofs 
+  between `FESpaces` defined on this model and its parent using a 
+  `RefinementTransferoperator`. 
+"""
+struct RefinedDiscreteModel{Dc,Dp,A<:DiscreteModel{Dc,Dp},B<:DiscreteModel{Dc,Dp},C<:RefinementGlue} <: DiscreteModel{Dc,Dp}
+  model  ::A
+  parent ::B
```

Comment:
To-think: do we actually want here to give the freedom to have `parent` to be potentially of a different type than `model`?
Why such a freedom is actually needed?

Another related question to-think: When we create a `RefinedDiscreteModel` (lets call it adapted model) out of another `RefinedDiscreteModel` (lets call it parent model), how do we intend to define the type of `A` and `B` in the adapted  model? 
1. `A=typeof(parent.model)` and `B=typeof(parent.parent)`?  
2. `A=typeof(parent.model)` and `B=RefinedDiscreteModel`? 
3. `A=RefinedDiscreteModel` and `B=typeof(parent.parent)`?
4. `A=RefinedDiscreteModel` and `B=RefinedDiscreteModel`?

I think that 1. is the most reasonable option, but not 100% sure ...

Reply:
I do not enforce that `model` and `parent` are of the same type because if we refine a `RefinedDiscreteModel` its son will have `typeof(parent) = RefinedDiscreteModel`. At least for now. Another question entirely would be if that is the way to go.

Reply:
As an example, we could consider 
```
model1 = CartesianDiscreteModel(...)
model2 = refine(model1)
model2 = refine(model2)
```
Then 
```
typeof(model1) = CartesianDiscreteModel

typeof(model2.model) = CartesianDiscreteModel
typeof(model2.parent) = CartesianDiscreteModel

typeof(model3.model) = CartesianDiscreteModel
typeof(model3.parent) = RefinedDiscreteModel
```

Reply:
Or this is how I had imagined it. We could definitely make it so that a refined model has its `.parent` be of type `parent.model` as you suggest. I actually like it, and I think it could be very good for multiple dispatch (for instance in the `refine()` function.

Reply:
Should I implement this change?

Reply:
> We could definitely make it so that a refined model has its .parent be of type parent.model as you suggest. I actually like it

But note that if we follow this approach, we would loose the possibility to walk up in the hierarchy. Not sure if this later feature can be needed at all.

Reply:
> we would loose the possibility to walk up in the hierarchy.

Actually this is not totally true in the sense that in the case of GMG preconditioning we have specialized types to represent the hieararchy, i.e., `MeshHierarchyLevel`, `MeshHierarchy`, etc.

Reply:
> Should I implement this change?

Let us think a little bit more.

Reply:
For future reference: We agreed that `RefinedDiscreteModel.model` should never be a refined model and that `RefinedDiscreteModel.parent` needs to be a refined model (except on the first refinement level) in order to be able to backwards-retrace the hierarchy from finest to coarsest. So this discussion is resolved.

Reply:
> We agreed that RefinedDiscreteModel.model should never be a refined model and that RefinedDiscreteModel.parent needs to be a refined model (except on the first refinement level) in order to be able to backwards-retrace the hierarchy from finest to coarsest. So this discussion is resolved.

Can you please add this to the docstring (if you did not yet)?